core/src/main/java/org/openmbee/mms/core/services/DefaultPermissionService.java

```diff
@@ -357,72 +331,33 @@ private Organization getOrganization(String orgId) {
         return org.get();
     }
 
-    private Project getProject(String projectId) {
-        Optional<Project> proj = projectRepo.findByProjectId(projectId);
+    private ProjectJson getProject(String projectId) {
+        Optional<ProjectJson> proj = projectPersistence.findById(projectId);
 
-        if (!proj.isPresent()) {
+        if (proj.isEmpty()) {
             throw new NotFoundException("Project " + projectId + " not found");
         }
         return proj.get();
     }
 
     private enum BRANCH_NOTFOUND_BEHAVIOR {THROW, CREATE, IGNORE}
 
-    private Branch getBranch(String projectId, String branchId, BRANCH_NOTFOUND_BEHAVIOR mode) {
-        Optional<Branch> branch = branchRepo.findByProject_ProjectIdAndBranchId(projectId, branchId);
-        if(!branch.isPresent()) {
-            switch(mode){
+    private RefJson getBranch(String projectId, String branchId, BRANCH_NOTFOUND_BEHAVIOR mode) {
+        Optional<RefJson> branch = branchPersistence.findById(projectId, branchId);
+        if (branch.isEmpty()) {
+            switch (mode) {
                 case THROW:
-                    throw new NotFoundException("Branch " +  projectId + " " + branchId + " not found");
+                    throw new NotFoundException("Branch " + projectId + " " + branchId + " not found");
                 case CREATE:
-                    Branch b = new Branch(getProject(projectId), branchId, false);
-                    branchRepo.save(b);
-                    return b;
+                    RefJson b = new RefJson();
```

Comment:
before was only creating global branch, check what side effects using persistence.save have

Reply:
fixed in #260